pkg/config/app_config.go

```diff
@@ -51,7 +53,32 @@ func NewAppConfig(name, version, commit, date string, buildSource string, debugg
 		return nil, err
 	}
 
-	userConfig, err := loadUserConfigWithDefaults(configDir)
+	defaultConfigFile := filepath.Join(configDir, "config.yml")
+	userConfigFiles := []string{defaultConfigFile}
+
+	if _, err := os.Stat(defaultConfigFile); err != nil {
+		if os.IsNotExist(err) {
+			file, err := os.Create(defaultConfigFile)
+			if err != nil {
+				if strings.Contains(err.Error(), "read-only file system") {
+					userConfigFiles = userConfigFiles[:0]
+				} else {
+					return nil, err
+				}
+			} else {
+				file.Close()
+			}
+		} else {
+			return nil, err
+		}
+	}
+
+	userConfigFilesOverwrite := os.Getenv("LG_CONFIG_FILE")
```

Comment:
I think we should be applying this override before the code above, so that we can handle errors with missing files etc. In fact I don't believe that validation logic needs to be moved from where it was

Reply:
fixed i think